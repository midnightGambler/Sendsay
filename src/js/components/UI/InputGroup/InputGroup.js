import React from "react";

const InputField = ({ label, inputProps }) => {
  return (
    <div className="input-group input-group-mb">
      <label className="input-group__label">{label}</label>
      <div className="input-group__inputs">
        {inputProps.map(({ error, ...rest }, key) => (
          <div key={key} className="input-group__inner-wrapper">
            <input className="input-group__inner-wrapper__input" {...rest} />
            {error && <span className="input-error">{error}</span>}
          </div>
        ))}
      </div>
    </div>
  );
};

export default InputField;
