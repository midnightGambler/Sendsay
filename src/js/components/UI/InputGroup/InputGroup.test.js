import React from "react";
import { shallow } from "enzyme";
import InputGroup from "./InputGroup";

it("should render component properly", () => {
  const inputGroup = shallow(
    <InputGroup
      inputProps={[{ name: "test" }, { name: "anotherTest" }]}
      label="test"
    />
  );
  expect(inputGroup).toMatchSnapshot();
});

it("should render component with error fields", () => {
  const inputGroup = shallow(
    <InputGroup
      label="test"
      inputProps={[{ error: "test" }, { error: "another test" }]}
    />
  );
  expect(inputGroup).toMatchSnapshot();
});

it("should test InputGroup onChange event", () => {
  const mockedCallback = jest.fn();
  const inputGroup = shallow(
    <InputGroup
      inputProps={[
        {
          onChange: mockedCallback
        }
      ]}
      label="test"
    />
  );
  inputGroup.find("input").simulate("change");
  expect(mockedCallback.mock.calls.length).toBe(1);
});
