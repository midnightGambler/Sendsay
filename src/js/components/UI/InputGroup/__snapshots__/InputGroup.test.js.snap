// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render component properly 1`] = `
<div
  className="input-group input-group-mb"
>
  <label
    className="input-group__label"
  >
    test
  </label>
  <div
    className="input-group__inputs"
  >
    <div
      className="input-group__inner-wrapper"
      key="0"
    >
      <input
        className="input-group__inner-wrapper__input"
        name="test"
      />
    </div>
    <div
      className="input-group__inner-wrapper"
      key="1"
    >
      <input
        className="input-group__inner-wrapper__input"
        name="anotherTest"
      />
    </div>
  </div>
</div>
`;

exports[`should render component with error fields 1`] = `
<div
  className="input-group input-group-mb"
>
  <label
    className="input-group__label"
  >
    test
  </label>
  <div
    className="input-group__inputs"
  >
    <div
      className="input-group__inner-wrapper"
      key="0"
    >
      <input
        className="input-group__inner-wrapper__input"
      />
      <span
        className="input-error"
      >
        test
      </span>
    </div>
    <div
      className="input-group__inner-wrapper"
      key="1"
    >
      <input
        className="input-group__inner-wrapper__input"
      />
      <span
        className="input-error"
      >
        another test
      </span>
    </div>
  </div>
</div>
`;
